src/tools/debugger/entities/SteppingType.java

```diff
@@ -148,34 +137,51 @@ public void process(final Suspension susp) {
     @Override
     public void process(final Suspension susp) {
       susp.getEvent().prepareContinue();
-      susp.getActivityThread().setSteppingStrategy(new AfterCommit());
+      susp.getActivityThread().setSteppingStrategy(this);
     }
   },
 
   @SerializedName("stepToMessageRcvr")
-  STEP_TO_RECEIVER_MESSAGE("stepToMessageRcvr", "Step to Msg Receiver", Group.ACTOR_STEPPING, "msg-open", new Class[] {EventualMessageSend.class}) {
-    @Override public void process(final Suspension susp) {
+  STEP_TO_MESSAGE_RECEIVER("stepToMessageRcvr", "Step to Msg Receiver",
+      Group.ACTOR_STEPPING, "msg-open", new Class[] {EventualMessageSend.class}) {
+    @Override
+    public void process(final Suspension susp) {
       susp.getEvent().prepareStepOver(1);
-      susp.getActivityThread().setSteppingStrategy(new ToMessageReceiver());
-      }
-    },
+      susp.getActivityThread().setSteppingStrategy(this);
+    }
+  },
+
+  @SerializedName("stepToPromiseResolver")
+  STEP_TO_PROMISE_RESOLVER("stepToPromiseResolver", "Step to Promise Resolver",
+      Group.ACTOR_STEPPING, "msg-white",
```

Comment:
I should probably find more intuitive icons for each stepping type, they have to be different for each command.